BrainPortal/public/javascripts/userfiles.js

```diff
@@ -722,25 +722,45 @@ $(function() {
         }
       }
 
-      if (window.FileReader && $('#upload-dialog').data('max-upload-size'))
-        /* check selected files against upload limit */
+      if (window.FileReader)
+        // Checks the file for problems, and disables the upload button if:
+        //   - no file selected
+        //   - spaces in filename
+        //   - filesize greater than maximum allowed by server
         $('#upload-dialog')
-          .undelegate('#up-file', 'change.uf.up-limit')
-          .delegate(  '#up-file', 'change.uf.up-limit', function () {
+          .undelegate('#up-file', 'change.uf.file-problem')
+          .delegate(  '#up-file', 'change.uf.file-problem', function () {
+            var warning_text = "";
             var max = parseInt($('#upload-dialog').data('max-upload-size'));
+            var bad_file;
+            var selected = !!$(this).val();
 
-            var visible = (
-              this.files && this.files[0] &&
-              max && this.files[0].size > max
-            );
+            var spaces_in_filename = $(this).val().includes(" ");
```

Comment:
Actually, why not compare it to the regex we use in the userfile model's validations ? That way we can complain about other illegal characters.